easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,167 @@
+##
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild. If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Code_Saturne,
+implemented as an easyblock.
+
+@author: Metin Cakircali (Juelich Supercomputing Centre, FZJ GmbH)
+"""
+
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig.default import CUSTOM
+from easybuild.tools.filetools import apply_regex_substitutions
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+import easybuild.tools.environment as env
+
+
+class EB_Code_underscore_Saturne(EasyBlock):
+    """Support for building and installing Code_Saturne."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to Code_Saturne."""
+        extra_vars = EasyBlock.extra_options(extra_vars)
+
+        # add some custom easyconfig parameters specific to Code_Saturne
+        extra_vars.update({
+            'debug': [False, "Build the debug version.", CUSTOM],
+            'slurm': [False, "Build for the slurm workload manager.", CUSTOM],
+        })
+
+        return extra_vars
+
+    def configure_step(self):
+        """Configure step for Code_Saturne."""
+
+        self.log.info("Configuration step is running...")
+
+        # only use the opt flags
+        env.setvar("CFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("CXXFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("FCFLAGS", os.environ['OPTFLAGS'])
```

Comment:
@mcakircali Can you clarify what this is about? This is counterintuitive w.r.t. what EasyBuild usually does.

Reply:
In this case, EB flags should add on top of default Code_Saturne flags except the optarch.

Reply:
* correction: "should not"

Reply:
Maybe there is a better way to control and suppress the EB flags. I admit that it's not flexible if you want to play with the flags, but that shouldn't be played around in these kind of solvers because they come with their own flags tested by developers.

Reply:
What other flags are you concerned about? Maybe these can be changed in the `toolchainopts` of the easyconfig instead? If so, looks like a better approach to me. If not, we'll have to evaluate how to do this.

Reply:
We can talk next week face-to-face for sure, but in any case we should report back here so there is a trace of why things are done in a particular way

Reply:
of course we will document here

Reply:
So what's the conclusion? Cause this indeed seems pretty weird.